docs/quickstart/run_example_quickstart.md

```diff
@@ -27,116 +27,360 @@ under the License.
+{% endhighlight %}
+
+This creates a `DataStream` of `WikipediaEditEvent` elements that we can further process. For
+the purposes of this example we are interrested in determining the number of added or removed
+bytes that each user causes in a certain time window, let's say five seconds. For this we first
+have to specify that we want to key the stream on the user name, that is to say that operations
+on this should take the key into account. In our case the summation of edited bytes in the windows
+should be per unique user. For keying a Stream we have to provide a `KeySelector`, like this:
+
+{% highlight java %}
+KeyedStream<WikipediaEditEvent, String> keyedEdits = edits
+    .keyBy(new KeySelector<WikipediaEditEvent, String>() {
+        @Override
+        public String getKey(WikipediaEditEvent event) {
+            return event.getUser();
+        }
+    });
+{% endhighlight %}
+
+This gives us the same Stream of `WikipediaEditEvent` that has a `String` key that is the user.
+We can now specify that we want to have windows imposed on this stream and compute some
+result based on elements in these windows. We need to specify a window here because we are
+dealing with an infinite stream of events. If you want to compute an aggregation on such an
+infinite stream you never know when you are finished. That's where windows come into play,
+they specify a time slice in which we should perform our computation. In our example we will say
+that we want to aggregate the sum of edited bytes for every five seconds:
+
+{% highlight java %}
+DataStream<Tuple2<String, Long>> result = keyedEdits
+    .window(TumblingProcessingTimeWindows.of(Time.seconds(5)))
+    .fold(new Tuple2<>("", 0L), new FoldFunction<WikipediaEditEvent, Tuple2<String, Long>>() {
+        @Override
+        public Tuple2<String, Long> fold(Tuple2<String, Long> acc, WikipediaEditEvent event) {
+            acc.f0 = event.getUser();
+            acc.f1 += event.getByteDiff();
+            return acc;
+        }
+    });
+{% endhighlight %}
+
+The first call, `.window()`, specified that we want to have tumbling (non-overlapping) windows
+of five seconds. The second call specifies a *Fold transformation* on each window slice for
+each unique key. In our case we start from an initial value of `("", 0L)` and add to it the byte
+difference of every edit in that time window for a user. The resulting Stream now contains
+a `Tuple2<String, Long>` for every user which gets emitted every five seconds.
+
+The only thing left to do is print the stream to the console and start execution:
+
+{% highlight java %}
+result.print();
+
+see.execute();
+{% endhighlight %}
+
+That last call is necessary to start the actual Flink job. All operations, such as creating
+sources, transformations and sinks only build up a graph of internal operations. Only when
+`execute()` is called is this graph of operations thrown on a cluster or executed on your local
```

Comment:
Shall we mention that the `print()`/`execute()` combination works differently in the batch case or would it make things complicated?